### ISSUE TITLE
Segment bug?

### ISSUE THREAD

**Issue by henry2004y** [JuliaGeometry/Meshes.jl#29]
Hi,

I am confused by the `segment` function results:
```
using Meshes

s1 = Segment((1.5,1.5), (3.0,1.5))
s2 = Segment((1.5,1.5), (3.5,1.5))
s3 = Segment((3.0,1.0), (2.0,2.0))

s1 ∩ s3  # Point(2.4, 1.5), wrong
s2 ∩ s3  # Point(2.5, 1.5), correct
```
Is it a bug? Sorry if I made a mistake here; too tired to think...

**Comment by henry2004y** [comment]
Sorry I didn't check; it was already reported #27.

**Comment by serenity4** [comment]
Hi, thanks for signaling the error. This is actually different. #30 fixes #27 but not this case.

**Comment by juliohm** [comment]
Thank you @henry2004y for reporting this 🐞. In this case the correct output of `s1 ∩ s3` should be `Point(2.5, 1.5)`.
